src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -2781,35 +2781,76 @@ kernel_arg_transaction_finalize (GObject *object)
 }
 
 static gboolean
-kernel_arg_transaction_execute (RpmostreedTransaction *transaction,
-                                GCancellable *cancellable,
-                                GError **error)
+kernel_arg_apply(KernelArgTransaction *self,
+                 RpmOstreeSysrootUpgrader *upgrader,
+                 OstreeKernelArgs *kargs,
+                 GCancellable *cancellable,
+                 GError **error)
 {
-  KernelArgTransaction *self = (KernelArgTransaction *) transaction;
-  OstreeSysroot *sysroot = rpmostreed_transaction_get_sysroot (transaction);
-  int upgrader_flags = 0;
-  auto command_line = static_cast<const char *>(vardict_lookup_ptr (self->options, "initiating-command-line", "&s"));
-  g_autofree char **append_if_missing = static_cast<char **>(vardict_lookup_strv_canonical (self->options, "append-if-missing"));
-  g_autofree char **delete_if_present = static_cast<char **>(vardict_lookup_strv_canonical (self->options, "delete-if-present"));
-  gboolean changed = FALSE;
+  g_return_val_if_fail (self != NULL, FALSE);
```

Comment:
In new or modified code let's prefer `g_assert()`.  See also https://github.com/ostreedev/ostree/pull/2463/commits/a8eed03a1974c285cb7ddf550c7385a4783bb97d

Reply:
This is a different case though. The `ostree` problem was about returning a `NULL` in places where consumers would not expect them (as per annotations). Here we do have proper "FALSE as error" everywhere, so no need to panic the whole daemon when a simpler error can be returned as usual.

Reply:
The thing is that returning `FALSE` *without* also setting the `GError` violates GError conventions and will (hopefully!) trip a later assertion.  There's no Rust equivalent for this...I guess it's a bit like `panic!(<undefined value>)`.

So we should either `g_assert()` or change these to return an error.  But since they're part of the internal code and not user facing, I think assert is right.

Reply:
(We can do this as a followup, or not)